hoomd/hpmc/integrate.py

```diff
@@ -399,14 +407,18 @@ def _attach_hook(self):
 
         HPMC uses RNGs. Warn the user if they did not set the seed.
         """
-        if (isinstance(self._simulation.device, hoomd.device.CPU)
-                and self._simulation.device.num_cpu_threads > 1
-                and any([f != 0 for f in self.depletant_fugacity.values()])):
-            warnings.warn(
-                "num_cpu_threads > 1 is deprecated since 4.4.0. "
-                "Use num_cpu_threads=1.",
-                FutureWarning,
-                stacklevel=2)
+        if any([f != 0 for f in self.depletant_fugacity.values()]):
+            warnings.warn("depletant_fugacity > 0 is deprecated since 4.4.0.",
+                          FutureWarning,
+                          stacklevel=1)
+
+            if (isinstance(self._simulation.device, hoomd.device.CPU)
+                    and self._simulation.device.num_cpu_threads > 1):
+                warnings.warn(
+                    "num_cpu_threads > 1 is deprecated since 4.4.0. "
+                    "Use num_cpu_threads=1.",
+                    FutureWarning,
+                    stacklevel=1)
```

Comment:
Shouldn't this warning be thrown also when depletant_fugacity is set to 0? Otherwise, I don't understand why we throw this warning at all if depletant_fugacity > 0 is deprecated anyway.

Reply:
In HPMC, only the implicit depletant code paths use TBB. The default behavior of TBB is set `num_cpu_threads` to the number of CPU cores on the system. By placing the `num_cpu_threads` warning inside the `depletant_fugacity > 0` check, I am preventing the `num_cpu_threads > 1` warning from appearing for all users of HPMC by default.

And yes, these warnings are redundant in this case. However, a user can act on the `num_cpu_threads` warning to future proof their code while still choosing to use the currently available, but deprecated, implicit depletant code path.